src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -726,6 +727,145 @@ public void writeMethodDefinition_withAnnotationsAndThrows() {
+    assertEquals(
+        writerVisitor.write(),
+        String.format(
+            createLines(6),
+            "package com.google.example.library.v1.stub;\n",
+            "\n",
+            "@Deprecated\n",
+            "@SuppressWarnings(\"all\")\n",
+            "public final class LibraryServiceStub extends String implements Appendable,"
+                + " Cloneable, Readable {\n",
+            "}"));
+  }
+
+  @Test
+  public void writeClassDefinition_statementsAndMethods() {
+    List<Statement> statements =
+        Arrays.asList(
+            ExprStatement.withExpr(
+                VariableExpr.builder()
+                    .setVariable(createVariable("x", TypeNode.INT))
+                    .setIsDecl(true)
+                    .setScope(ScopeNode.PRIVATE)
+                    .build()),
+            ExprStatement.withExpr(
+                VariableExpr.builder()
+                    .setVariable(createVariable("y", TypeNode.INT))
+                    .setIsDecl(true)
+                    .setScope(ScopeNode.PROTECTED)
+                    .build()));
+
+    MethodDefinition methodOne =
+        MethodDefinition.builder()
+            .setName("open")
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(TypeNode.BOOLEAN)
+            .setReturnExpr(
+                ValueExpr.builder()
+                    .setValue(
+                        PrimitiveValue.builder().setType(TypeNode.BOOLEAN).setValue("true").build())
+                    .build())
+            .build();
+
+    MethodDefinition methodTwo =
+        MethodDefinition.builder()
+            .setName("close")
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(TypeNode.VOID)
+            .setBody(
+                Arrays.asList(
+                    ExprStatement.withExpr(
+                        createAssignmentExpr("foobar", "false", TypeNode.BOOLEAN))))
+            .build();
+
+    List<MethodDefinition> methods = Arrays.asList(methodOne, methodTwo);
+
+    ClassDefinition nestedClassDef =
+        ClassDefinition.builder()
+            .setName("IAmANestedClass")
+            .setIsNested(true)
+            .setScope(ScopeNode.PRIVATE)
+            .setIsStatic(true)
+            .setMethods(Arrays.asList(methodOne))
+            .build();
+
+    ClassDefinition classDef =
+        ClassDefinition.builder()
+            .setPackageString("com.google.example.library.v1.stub")
+            .setName("LibraryServiceStub")
+            .setScope(ScopeNode.PUBLIC)
+            .setStatements(statements)
+            .setMethods(methods)
+            .setNestedClasses(Arrays.asList(nestedClassDef))
+            .build();
+
+    classDef.accept(writerVisitor);
+    assertEquals(
+        writerVisitor.write(),
+        String.format(
+            createLines(18),
+            "package com.google.example.library.v1.stub;\n",
+            "\n",
+            "public class LibraryServiceStub {\n",
+            "private int x;\n",
```

Comment:
Are we consider a tab space inside of class.

Reply:
No, we don't need to indent this because google-java-format will take care of it.